cli/lib/kontena/cli/common.rb

```diff
@@ -79,96 +185,132 @@ def ensure_custom_ssl_ca(url)
+        config.current_master
       end
 
       def current_master=(master_alias)
-        settings['current_server'] = master_alias
-        save_settings
+        config.current_master = master_alias
       end
 
-      def error(message = nil)
-        $stderr.puts(message) if message
+      def error(message = "Error")
+        prompt.error(message)
         exit(1)
       end
 
-      def prompt(prefix = '> ')
-        require 'highline/import'
-        ask(prefix)
+      def ask(question = "")
+        prompt.ask(question)
+      end
+
+      def yes?(question = "")
+        prompt.yes?(question)
+      end
+
+      def prompt
+        ::Kontena.prompt
       end
 
       def confirm_command(name, message = nil)
+        if self.respond_to?(:force?) && self.force?
+          return
+        end
         puts message if message
         puts "Destructive command. To proceed, type \"#{name}\" or re-run this command with --force option."
 
-        prompt == name || error("Confirmation did not match #{name}. Aborted command.")
+        ask("Enter '#{name}' to confirm: ") == name || error("Confirmation did not match #{name}. Aborted command.")
       end
 
-      def confirm(message = 'Destructive command. Are you sure? (y/n) or re-run this command with --force option.')
-        puts message
-
-        ['y', 'yes'].include?(prompt) || error("Aborted command.")
+      def confirm(message = 'Destructive command. You can skip this prompt by running this command with --force option. Are you sure?')
+        if self.respond_to?(:force?) && self.force?
+          return
+        end
+        prompt.yes?(message) || error('Aborted command.')
       end
 
       def api_url=(api_url)
-        settings['servers'][current_master_index]['url'] = api_url
-        save_settings
+        config.current_master.url = api_url
+        config.write
       end
 
       def access_token=(token)
-        settings['servers'][current_master_index]['token'] = token
-        save_settings
+        require_current_master.token.access_token = token
+        config.write
       end
 
       def add_master(server_name, master_info)
-        server_name = server_name || 'default'
-        index = settings['servers'].find_index{|m| m['name'] == server_name}
-        if index
-          settings['servers'][index] = master_info
-        else
-          settings['servers'] << master_info
+        config.add_server(master_info.merge('name' => server_name))
+      end
+
+      def spinner(msg, &block)
+        Kontena::Cli::Spinner.spin(msg, &block)
+      end
+
+      def any_key_to_continue
+        return nil if running_silent?
+        msg = "Press any key to continue or ctrl-c to cancel.. "
+        print pastel.bright_white("#{msg}")
+        char = STDIN.getch
+        print "\r#{' ' * msg.length}\r"
+        if char == "\u0003"
+          error "Canceled"
+        end
+      end
+
+      def display_login_info(only: nil)
+        unless only == :account
```

Comment:
these can be refactored into own methods for example `display_account_master_info` etc.